### DISCUSSION TITLE
add some Important function

### DISCUSSION THREAD

**Discussion by p82183645** [Zrips/CMI#1920] [Feature Requests]
1. add /cmi data , Change item data
Uses: Convenient to make material package item

Example: DIAMOND_SWORD:5
/cmi data 6
change
 DIAMOND_SWORD:6

2. add Maximum enchantment limit
Example: Efficiency III
The server does not appear to exceed the efficiency III

3. armor_stand edit gui , add remove Button ( Sometimes armor can't be removed )

4. /cmi reload , can reload SavedItems.yml 
Manual editing is more convenient
or add /cmi SavedItems reload

5. Kit condition , add play time condition
Example: This player has been playing for 7 days in total to receive this kit.

thank , please add
Especially 1 and 4
These two functions are very important

**Comment by p82183645** [comment]
Please add
I have requested many times

**Comment by Dracathio** [comment]
Alright how to answer this well...
1. No clue what this is about, but I'm sure it could be something not in the plugin yet.
2. If you even paid bloody attention to the config.yml you would know near the very bottom THERE ARE SETTINGS FOR ALL ENCHANTS AND THEIR LIMITATIONS IN SERVER.
3. It's a shift click basis or something that can also be found in the config I am sure.
4. Eh.. that would mean setting an entire thing up that asks what should be reloaded on /cmi reload
5. Already possible, you really haven't looked into things yet about playtime rewards at all.  Access to the kit or handing the kit out is something you're able to do if you make it to where the play over time rewards {PlayTimeRewards.yml} do something like lp user [playername] parent add [groupname] and give that group the permission node for that kit.

so... about 3 of these ARE IN THE PLUGIN you're just asking for him to make it easier when it's just you not looking into the plugin hard enough to find all the different settings.  Which I do understand, because the plugin is quite hard to navigate.

**Comment by p82183645** [comment]
> Alright how to answer this well...
> 
> 1. No clue what this is about, but I'm sure it could be something not in the plugin yet.
> 2. If you even paid bloody attention to the config.yml you would know near the very bottom THERE ARE SETTINGS FOR ALL ENCHANTS AND THEIR LIMITATIONS IN SERVER.
> 3. It's a shift click basis or something that can also be found in the config I am sure.
> 4. Eh.. that would mean setting an entire thing up that asks what should be reloaded on /cmi reload
> 5. Already possible, you really haven't looked into things yet about playtime rewards at all.  Access to the kit or handing the kit out is something you're able to do if you make it to where the play over time rewards {PlayTimeRewards.yml} do something like lp user [playername] parent add [groupname] and give that group the permission node for that kit.
> 
> so... about 3 of these ARE IN THE PLUGIN you're just asking for him to make it easier when it's just you not looking into the plugin hard enough to find all the different settings. Which I do understand, because the plugin is quite hard to navigate.

1. 
I said very clearly, change the data of the item.
Use for making resource package items

2.
That is not mandatory to limit all enchants
That is for the "command" enchantment

4.
I said very clearly
/cmi saveitem reload
Used to manually edit files
This is very convenient!

**Comment by Zrips** [comment]
@p82183645 
1. I'm confused with this point too. There is so called NBT data which one can be seen with itemnbt command. Are we talking about this data?
2. This one would require some heavy lifting with reflections and stuff, to change base enchant max value. Possible tho. 
3. Some armor stands can be none intractable, this means you have to change this with armorstand editor. /cmi armorstand near and click on Interactable button and change it to green.
4. SavedItems will reload properly with next build when using /cmi reload
5. For kit conditions as player playtime, well that one could be added

**Comment by p82183645** [comment]
> @p82183645
> 
> 1. I'm confused with this point too. There is so called NBT data which one can be seen with itemnbt command. Are we talking about this data?
> 2. This one would require some heavy lifting with reflections and stuff, to change base enchant max value. Possible tho.
> 3. Some armor stands can be none intractable, this means you have to change this with armorstand editor. /cmi armorstand near and click on Interactable button and change it to green.
> 4. SavedItems will reload properly with next build when using /cmi reload
> 5. For kit conditions as player playtime, well that one could be added

1.
Example: 
 Diamond _sword:5 , use /cmi data 6 , change to Diamond _sword:5
 Diamond _sword:10 , use /cmi data 10 , change to Diamond _sword:10

Use for making resource package items

**Comment by p82183645** [comment]
@Zrips  hallo?

**Comment by Dracathio** [comment]
how about you get a plugin that does that for you...

I mean what are you even trying to do?

If you're trying to change the strength or such of a diamond sword... then use a plugin that can do that.

Also you're wrong about enchantment limitation.  I can go and do /enchant 10 to anything I wish even if the limitation for enchanting is set to 5.  The enchant limit is for items you're trying to enchant without using a command.  How do I know this?  because I've been using CMI for half a year... I enchanted items with CMI that went over the limited enchant numbers.

get the free version of mythic mobs to do what you wish to do with your diamond swords, if you're trying to do something related to texture packs... then find a plugin that helps with that, because cmi may not be the one that will do that.  Also if you really need a diamond sword that badly with a certain data value... use command blocks to get it. or a command or what ever... because in all honesty everything a plugin does.... is pretty much doable with command blocks in some manner.
